File: core/src/main/java/me/devsnox/planetsystem/core/PlanetSystem.java
package me.devsnox.planetsystem.core;

import me.devsnox.planetsystem.api.planet.LoadedPlanet;
import me.devsnox.planetsystem.api.player.PlanetPlayer;
import me.devsnox.planetsystem.core.commands.PlanetCommand;
import me.devsnox.planetsystem.core.holder.HolderImpl;
import me.devsnox.planetsystem.core.listeners.PlanetListener;
import me.devsnox.planetsystem.core.listeners.PlayerListener;
import org.bukkit.Bukkit;
import org.bukkit.plugin.PluginManager;
import org.bukkit.plugin.java.JavaPlugin;

import java.util.Set;
import java.util.logging.Level;

import static me.devsnox.planetsystem.api.holder.Holder.Impl.holder;

public class PlanetSystem extends JavaPlugin
{
    @Override
    public void onEnable()
    {
        holder = new HolderImpl(this);
        this.registerListeners();
        this.registerCommands();
//        Logger.getLogger("org.mongodb").setLevel(Level.SEVERE);
//        Logger.getLogger("org.mongodb.driver").setLevel(Level.WARNING);
//
//        final LoggerContext loggerContext = (LoggerContext) LoggerFactory.getILoggerFactory();
//        loggerContext.getLogger("org.mongodb.driver").setLevel(ch.qos.logback.classic.Level.OFF);
        this.getLogger().log(Level.INFO, "PlanetSystem started");
    }
    
    @Override
    public void onDisable()
    {
        this.saveAll();
        this.getLogger().log(Level.INFO, "PlanetSystem stopped");
    }
    
    public void saveAll()
    {
        final Set<LoadedPlanet> planets = holder.getPlanetData().getLoadedPlanets();
        final Set<PlanetPlayer> players = holder.getPlayerData().getPlayers();
        if (planets.size() != 0)
        {
            planets.forEach(planet -> holder.getPlanetData().save(planet.getOwnerUniqueID()));
        }
        if (players.size() != 0)
        {
            players.forEach(player -> holder.getPlayerData().save(player.getUUID()));
        }
    }
    
    private void registerCommands()
    {
        this.getCommand("planet").setExecutor(new PlanetCommand());
    }
    
    private void registerListeners()
    {
        final PluginManager pluginManager = Bukkit.getPluginManager();
        pluginManager.registerEvents(new PlanetListener(), this);
        pluginManager.registerEvents(new PlayerListener(), this);
    }
}

File: core/src/main/kotlin/me/devsnox/planetsystem/core/PlanetSystem.kt
package me.devsnox.planetsystem.core

import me.devsnox.planetsystem.api.holder.Holder.Impl.holder
import me.devsnox.planetsystem.core.commands.PlanetCommand
import me.devsnox.planetsystem.core.holder.HolderImpl
import me.devsnox.planetsystem.core.listeners.PlanetListener
import me.devsnox.planetsystem.core.listeners.PlayerListener
import org.bukkit.Bukkit
import org.bukkit.plugin.java.JavaPlugin


class PlanetSystem : JavaPlugin() {

	override fun onLoad() {
		holder = HolderImpl()
	}

	override fun onEnable() {
		register()
		logger.info("PlanetSystem started")
	}

	override fun onDisable() {
		saveAll()
		logger.info("PlanetSystem stopped")
	}

	private fun saveAll() {
		holder.planetData.loadedPlanets.forEach { holder.planetData.save(it.ownerUniqueID) }
		holder.playerData.players.forEach { holder.playerData.save(it.uuid) }
	}

	private fun register() {
		registerCommands()
		registerListeners()

		//TODO: Add config handling
		Bukkit.getScheduler().runTaskTimer(this, { saveAll() }, 0, 20 * 60)
	}

	private fun registerCommands() {
		PlanetCommand(this)
	}

	private fun registerListeners() {
		PlanetListener(this)
		PlayerListener(this)
	}

}
